Include/internal/pycore_object.h

```diff
@@ -58,6 +58,9 @@ extern void _Py_DecRefTotal(PyInterpreterState *);
 // Increment reference count by n
 static inline void _Py_RefcntAdd(PyObject* op, Py_ssize_t n)
 {
+    if (_Py_IsImmortal(op)) {
+        return;
+    }
```

Comment:
One quick comment here is that if this were to be in a hot path, I would recommend using the same technique that we use for `Py_INCREF` to minimize the number of generated instructions to do the check and add: https://github.com/python/cpython/blob/main/Include/object.h#L620-L634

Now, given that it's only in use for `sq_repeat` in tuple and list, I don't think this is perf sensitive so keeping the code as you have it here should be good.